src/tiledb/cloud/groups.py

```diff
@@ -236,3 +242,70 @@ def _add_to(*, namespace: str, name: str, parent_uri: str) -> None:
             ),
         ),
     )
+
+
+def list_shared_with(uri, async_req=False):
+    """List a group's sharing policies.
+
+    :param str uri: tiledb URI of the asset.
+    :param async_req: return future instead of results for async support.
+    :return: a list of GroupSharing objects.
+    """
+    (group_namespace, group_name) = split_uri(uri)
+    api_instance = client.build(rest_api.GroupsApi)
+
+    try:
+        return api_instance.get_group_sharing_policies(
+            group_namespace=group_namespace, group_name=group_name, async_req=async_req
+        )
+    except GenApiException as exc:
+        raise tiledb_cloud_error.check_exc(exc) from None
+
+
+def share_group(uri, namespace, permissions, async_req=False):
+    """Shares group with given namespace and permissions.
+
+    :param str uri: tiledb URI of the asset.
+    :param str namespace:
+    :param list(str) permissions:
+    :param async_req: return future instead of results for async support.
+    :return: None.
+    """
+
+    if not isinstance(permissions, list):
+        permissions = [permissions]
+
+    if set([perm.lower() for perm in permissions]) - {
+        models.GroupActions.READ,
+        models.GroupActions.WRITE,
+    }:
```

Comment:
I used a set comparison here instead of a loop over individual permissions. A minor optimization.